### PR TITLE
(MODULES-1999) Add AppVeyor

### PR DESCRIPTION
**Pull request by ferventcoder**
Add AppVeyor to acl module.

- Matrix - Run against several supported versions of Puppet
- Set fast finish on the matrix to fail immediately if one cell fails
- Notify the Ruby / Puppet versions
- Capture Environment vars / Gemfile.lock contents
- Ensure Administrator - AppVeyor doesn't have an Administrator account by
  default, so put the adminstrator account back for specs to pass
- create top level Puppet data folders - puppet runs on Windows fails until the
  folders exist
- Skip commits that are docs/acceptance tests
- clone only the last 10 commits
- turn off notifications

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    RUBY_VER: 200
+  - PUPPET_GEM_VERSION: 4.0.0
+    RUBY_VER: 215
+install:
+- SET PATH=C:\Ruby%RUBY_VER%\bin;%PATH%
+- gem install bundler --quiet --no-ri --no-rdoc
+- bundle install --without development
+build_script:
+- 'bundle exec puppet apply -e "user {''Administrator'': ensure=>present, groups=>[''Administrators''],}"'
+- exit 0
```
Comment (ferventcoder) [inline appveyor.yml]: @MikaelSmith Here you go

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+init:
+- SET
+environment:
+  matrix:
+  - PUPPET_GEM_VERSION: 3.4.3
+    RUBY_VER: 193
+  - PUPPET_GEM_VERSION: 3.7.5
+    RUBY_VER: 200
+  - PUPPET_GEM_VERSION: 4.0.0
+    RUBY_VER: 215
```
Comment (ferventcoder) [inline appveyor.yml]: A fun little matrix here for testing against multiple versions of Puppet

**Comment by ferventcoder** [comment]
This is building https://ci.appveyor.com/project/puppetlabs/puppetlabs-acl/build/1.1.x.14

**Comment by ferventcoder** [comment]
Found out that `Ruby 2.1.x` is NOT in the `c:\Ruby215` folder. It's in `c:\Ruby21` - @MikaelSmith we'll need to change this on Puppet.

**Comment by ferventcoder** [comment]
https://github.com/puppetlabs/puppet/pull/3885

**Comment by MikaelSmith** [comment]
Why doesn't this have AppVeyor status?

**Comment by ferventcoder** [comment]
You are gonna laugh - the commit message includes the words "docs/acceptance tests"

**Comment by ferventcoder** [comment]
That should make it kick off

**Comment by ferventcoder** [comment]
@MikaelSmith @e @joshcooper @cyberious this is ready for merge.

File: Gemfile [hunk 3]
```diff
@@ -10,9 +10,14 @@ def location_for(place, fake_version = nil)
...
-gem "puppet", *location_for(ENV['PUPPET_LOCATION'] || '~> 3.7.0')
-gem "facter", *location_for(ENV['FACTER_LOCATION'] || '~> 2.0')
-gem "hiera", *location_for(ENV['HIERA_LOCATION'] || '~> 1.0')
+if puppetversion = ENV['PUPPET_GEM_VERSION']
+  gem 'puppet', puppetversion,  :require => false
+else
+  gem 'puppet', *location_for(ENV['PUPPET_LOCATION'] || '~> 3')
+end
+
+gem "facter", *location_for(ENV['FACTER_LOCATION'] || '>= 1')
```
Comment (MikaelSmith) [inline Gemfile]: Does Facter 1 work?

Comment (ferventcoder) [inline Gemfile]: Facter 1.7.6 was in 3.4.3 of Puppet.

File: appveyor.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    RUBY_VER: 21-x64
+matrix:
+  fast_finish: true
+install:
+- SET PATH=C:\Ruby%RUBY_VER%\bin;%PATH%
+- gem install bundler --quiet --no-ri --no-rdoc
+- bundle install --jobs 4 --retry 2 --without development
+- type Gemfile.lock
+build_script:
+- 'bundle exec puppet apply -e "user {''Administrator'': ensure=>present, password => ''Th!s1sS#cur3@@@2129021'', groups=>[''Administrators''],}" && exit 0'
```
Comment (MikaelSmith) [inline appveyor.yml]: Tests will fail if this fails, so why suppress the error?

File: Gemfile (hunk 3 above)
Comment (ferventcoder) [inline Gemfile]: It's mostly handled with the Puppet Gem install, but when you are faking the locations, Facter is not automatically pulled in as a gem. Same with Hiera.

File: appveyor.yml (hunk 4 above)
Comment (ferventcoder) [inline appveyor.yml]: Good question. I think I was also getting an error not just in creation. It was last week though, it's likely that the suppression could be removed.

File: appveyor.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
+version: 1.1.x.{build}
+skip_commits:
+  message: /(^\(?doc\)?.*|.*[A|a]cceptance [T|t]est.*)/
+clone_depth: 10
+init:
+- SET
+- 'mkdir C:\ProgramData\PuppetLabs\code && exit 0'
```
Comment (MikaelSmith) [inline appveyor.yml]: Do these need to use `-Path`, or does that happen by default?

Comment (ferventcoder) [inline appveyor.yml]: Happens by default.

**Comment by MikaelSmith** [comment]
I'm fine merging as-is. The `exit 0` after calling `puppet` shouldn't create any issues.

**Comment by ferventcoder** [comment]
@cyberious @Iristyle I don't think that @MikaelSmith has access to do the merge, so if you are also good, let's move this one forward.

**Comment by Iristyle** [comment]
:+1:

**Pull request merged**